### DISCUSSION TITLE
How to serialize elements

### DISCUSSION THREAD

**Discussion by brillout** [solidjs/solid#1809] [Q&A] [answered]
How can I serialize elements to a string?

```tsx
const el = <p>Hello</p>
const str: string = serialize(el)
assert(str==='{"element":"p","children":[{"text":"Hello"}]}') // For example
function serialize(el: JSX.Element): string {
  // How?
}
```

This would allow us to pass elements from the server to the client.

## Context

[Vite-plugin-ssr](https://vite-plugin-ssr.com/) (I'm its author) is serializing React elements using [this](https://github.com/zasource-dev/React-Serialize/blob/master/src/index.js) and I'd like to offer the same capability to Solid users.

**Comment by atk** [comment, marked as answer]
Solid does not have a vDOM that simplifies serialization. Even more, components are mostly not static objects that are replaced with other static objects in the next render cycle, but they are mainly templates and reactive side effects. The components are not aware of their lifecycle either (unless you make them be manually). Thus, this is a task that, while it looks simple and straight forward, is incredibly hard, if not even almost impossible to get all the edge cases right for.

**Reply by brillout** [reply]
I see, makes sense, since Solid does "lazy-rendering".
